compiler/compiler/src/compiler.rs

```diff
@@ -172,22 +172,64 @@ impl<'a, N: Network> Compiler<'a, N> {
         ))
     }
 
+    /// Run const propagation and loop unrolling until we hit a fixed point or find an error.
+    pub fn const_propagation_and_unroll_loop(&mut self, symbol_table: &mut SymbolTable) -> Result<()> {
```

Comment:
Does it make sense to add some heuristic bound to the build options so that we can terminate the loop?